Lib/pydoc.py

```diff
@@ -488,28 +488,33 @@ def fail(self, object, name=None, *args):
     def getdocloc(self, object, basedir=sysconfig.get_path('stdlib')):
         """Return the location of module docs or None"""
 
+        if not inspect.ismodule(object):
+            return None
+
+        try:
+            from pydoc_data.topics import modules
+        except ImportError:
+            return None
+
+        if object.__name__ not in modules:
+            return None
+
         try:
             file = inspect.getabsfile(object)
         except TypeError:
             file = '(built-in)'
 
+        if 'site-packages' in file:
```

Comment:
Debian and derivative OSes also have `'dist-packages'` — not sure if that should be handled here upstream, or documented somehow to let the debian python maintainers know to patch this file.

Reply:
In fact, if we find the object's name in the `modules` dict, it's either a stdlib module (`getdocloc` is only called on modules) or something shadowing one.  I'm not sure the latter case is worth trying to detect, so I simplified.